### PR TITLE
add flash monitor baud rate (VSC-476)

### LINKED ISSUE #207: [Feature Request]: Configurable flash baudrate (VSC-473)
**Background**
It seems to me that when I use the ESP-IDF: Flash command (i.e. <kbd>Ctrl</kbd> + <kbd>E</kbd>  <kbd>F</kbd>) it gets the flash baud rate (the parameter passed to `-b`) from the configured device baudrate. I would like to flash at the fastest possible baud rate (i.e. 2MHz), but would not always want to communicate at that speed (prefer 115200 for comms).

```
> Executing task in folder sensor_platform: /home/cal/.espressif/python_env/idf4.3_py2.7_env/bin/python /home/cal/esp/esp-idf/components/esptool_py/esptool/esptool.py -p /dev/ttyUSB0 -b 2000000 --after hard_reset write_flash --flash_mode dio --flash_freq 40m --flash_size detect 0x8000 partition_table/partition-table.bin 0xe000 ota_data_initial.bin 0x1000 bootloader/bootloader.bin 0x10000 sensor_platform.bin <
```

**Preferred Solution**
A separate configurable parameter in the esp-idf config, picked up by the vscode extension.

**Alternatives Considered**
I tried using the workspace's `tasks.json` and modifying this, but it also seems to me that this is Task: Flash, and not the ESP-IDF: Flash command, so it did not override the built-in method.

I ended up modifying the extension's `/dist/extension.js` on my install and replacing `this.model.baudRate` with a fixed value of a higher baud rate.

**Additional context**
I realize that the configurable parameter would likely be rooted in the esp-idf config, but thought it would be best to start here.
I also note that changing the flash mode, from DIO to QIO in the esp config, did not seem to have an effect on the parameter passed to `--flash_mode` when flashing via the extension command.

### PR DESCRIPTION
**Pull request by brianignacio5**
Add Monitor and Flash baud rate configuration settings for vscode with higher default baud rate.

Fix #207

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -1385,7 +1397,7 @@ export async function activate(context: vscode.ExtensionContext) {
   );
   registerIDFCommand("espIdf.launchWSServerAndMonitor", async () => {
     const port = idfConf.readParameter("idf.port") as string;
-    const baudRate = idfConf.readParameter("idf.baudRate") as string;
+    const baudRate = idfConf.readParameter("idf.monitorBaudRate") as string;
```
Comment (pwmb) [inline src/extension.ts:1400]: rename the variables as well!

**Review by pwmb — COMMENTED** [review]
Maybe we can use [`CONFIG_ESPTOOLPY_MONITOR_BAUD `](https://github.com/espressif/vscode-esp-idf-extension/issues/37#issuecomment-697306183) as suggested by @igrr 

Also for commands like `idf.py flash monitor`... what shall be the default behaviour ... for user who are setting both the flash baud rate and monitor baud rate they want both to be run in their own baud rate ... right? is this case handled?

**Comment by brianignacio5** [comment]
> Maybe we can use [`CONFIG_ESPTOOLPY_MONITOR_BAUD `](https://github.com/espressif/vscode-esp-idf-extension/issues/37#issuecomment-697306183) as suggested by @igrr
> 
> Also for commands like `idf.py flash monitor`... what shall be the default behaviour ... for user who are setting both the flash baud rate and monitor baud rate they want both to be run in their own baud rate ... right? is this case handled?

For `idf.py flash monitor` I assume you mean in the ESP-IDF Terminal, where the user can specify the baud rate which is probably going to be use for both flash and monitor actions.

**Comment by pwmb** [comment]
> I assume you mean in the ESP-IDF Terminal, where the user can specify the baud rate

No I mean from the command `espIdf.buildFlashMonitor`

**Comment by brianignacio5** [comment]
> > I assume you mean in the ESP-IDF Terminal, where the user can specify the baud rate
> 
> No I mean from the command `espIdf.buildFlashMonitor`

buildFlashMonitor has `idf.flashBaudRate` for the flash step and `idf.monitorBaudRate` for the monitor step.

File: templates/.vscode/tasks.json [hunk 2]
```diff
@@ -142,9 +142,9 @@
         {
             "label": "Monitor: Start the monitor",
             "type": "shell",
-            "command": "${config:idf.pythonBinPath} ${config:idf.espIdfPath}/tools/idf.py -p ${config:idf.port} -b ${config:idf.baudRate} monitor",
+            "command": "${config:idf.pythonBinPath} ${config:idf.espIdfPath}/tools/idf.py -p ${config:idf.port} monitor",
             "windows": {
-                "command": "${config:idf.pythonBinPathWin} ${config:idf.espIdfPathWin}\\tools\\idf.py monitor -p ${config:idf.portWin} -b ${config:idf.baudRate}"
+                "command": "${config:idf.pythonBinPathWin} ${config:idf.espIdfPathWin}\\tools\\idf.py monitor -p ${config:idf.portWin}"
```
Comment (pwmb) [inline templates/.vscode/tasks.json:147]: maybe we need to pass the monitor baud rate

**Review by pwmb — APPROVED** [review]
Thanks @brianignacio5 LGTM...

Left one minor comment

File: templates/.vscode/tasks.json (hunk 2 above)
Comment (brianignacio5) [reply templates/.vscode/tasks.json:147]: Based on @igrr comment, `idf.py monitor` already picks the value from CONFIG_ESPTOOLPY_MONITOR_BAUD in https://github.com/espressif/esp-idf/blob/8bc19ba893e5544d571a753d82b44a84799b94b1/tools/idf_py_actions/serial_ext.py#L87-L88

Comment (pwmb) [reply templates/.vscode/tasks.json:147]: I see maybe add a doc explaining this also add the priority/preference order of the baud rates for the monitor.

**Pull request merged**